### PR TITLE
feat(coordinator): add blob bytes in batch proving task

### PR DESCRIPTION
**Pull request by colinlyguo**
### PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [x] feat: A new feature

### CODE REVIEW

File: coordinator/internal/orm/batch.go [hunk 1]
```diff
@@ -31,6 +31,7 @@ type Batch struct {
 	WithdrawRoot    string `json:"withdraw_root" gorm:"column:withdraw_root"`
 	ParentBatchHash string `json:"parent_batch_hash" gorm:"column:parent_batch_hash"`
 	BatchHeader     []byte `json:"batch_header" gorm:"column:batch_header"`
+	EnableEncode    bool   `json:"enable_encode" gorm:"column:enable_encode"`
```
Comment (zimpha) [inline coordinator/internal/orm/batch.go]: can this be derived from version in `BatchHeader`?

base on the logic in function `getBatchTaskDetail`, if `EnableEncode=false`, you will use `codecv3.NewDABatch` and if `EnableEncode=true`, you will use `codecv4.NewDABatch(batchEncoding, true)`.

Doesn't it means if batch header version is 3, use `codecv3.NewDABatch`, if batch header version is 4, use `codecv4.NewDABatch(batchEncoding, true)`.

Comment (colinlyguo) [reply coordinator/internal/orm/batch.go]: Yes. Fixed in [9cb1ac5](https://github.com/scroll-tech/scroll/pull/1485/commits/9cb1ac59ae859084c316ac1116335c330bf385d6).

using: _if batch header version is 3, use codecv3.NewDABatch, if batch header version is 4, use codecv4.NewDABatch(batchEncoding, true)_

**Review by zimpha — APPROVED** [review]

File: coordinator/internal/logic/provertask/batch_prover_task.go [hunk 2]
```diff
@@ -241,3 +235,65 @@ func (bp *BatchProverTask) recoverActiveAttempts(ctx *gin.Context, batchTask *or
...
+	}
+
+	if encoding.CodecVersion(dbBatch.CodecVersion) == encoding.CodecV3 {
+		daBatch, createErr := codecv3.NewDABatch(batch)
+		if createErr != nil {
+			return nil, fmt.Errorf("failed to create DA batch (v3) for batch %d: %w", dbBatch.Index, createErr)
+		}
+		taskDetail.BlobBytes = daBatch.BlobBytes()
+
+		batchHeader, decodeErr := codecv3.NewDABatchFromBytes(dbBatch.BatchHeader)
```
Comment (georgehao) [inline coordinator/internal/logic/provertask/batch_prover_task.go:257]: Why don't you get `dbBatch.BatchHeader` from `orm.Batch.BatchHeader` ?

Comment (colinlyguo) [reply coordinator/internal/logic/provertask/batch_prover_task.go:257]: This is got from `orm.Batch.BatchHeader`, and the type fetched from db is `byte array`, so change it to `codecv3.Batch` by `codecv3.NewDABatchFromBytes`.

File: coordinator/internal/logic/provertask/batch_prover_task.go [hunk 3]
```diff
@@ -241,3 +235,65 @@ func (bp *BatchProverTask) recoverActiveAttempts(ctx *gin.Context, batchTask *or
...
+	chunks := make([]*encoding.Chunk, len(dbChunks))
+	for i, c := range dbChunks {
+		blocks, getErr := bp.blockOrm.GetL2BlocksInRange(ctx, c.StartBlockNumber, c.EndBlockNumber)
+		if getErr != nil {
+			return nil, fmt.Errorf("failed to get blocks in range for batch %d, chunk %d, (start: %d, end: %d): %w", dbBatch.Index, c.Index, c.StartBlockNumber, c.EndBlockNumber, getErr)
+		}
+		chunks[i] = &encoding.Chunk{Blocks: blocks}
+	}
+
+	dbParentBatch, getErr := bp.batchOrm.GetBatchByIndex(ctx, dbBatch.Index-1)
```
Comment (georgehao) [inline coordinator/internal/logic/provertask/batch_prover_task.go]: imagine the first batch is codec v4 batch, index=1, if you get the parentBatch is like this, will get a error

Comment (colinlyguo) [reply coordinator/internal/logic/provertask/batch_prover_task.go]: the parent batch (index=0) would be the genesis batch, which has a codec version 0. so no problem here.

Comment (colinlyguo) [reply coordinator/internal/logic/provertask/batch_prover_task.go]: added a check: [2bd9438](https://github.com/scroll-tech/scroll/pull/1485/commits/2bd943860dafbed119f6076338da90d306b33ea9).

File: coordinator/internal/logic/provertask/batch_prover_task.go (hunk 2 above)
Comment (colinlyguo) [reply coordinator/internal/logic/provertask/batch_prover_task.go:257]: changed to use db value in [bb4d3eb](https://github.com/scroll-tech/scroll/pull/1485/commits/bb4d3eb87ba75615a8a4126e8b83a99cd4d7523a).

**Pull request merged**